### PR TITLE
[DPoP] unit tests for util class

### PR DESCRIPTION
**Pull request by PasinduLakshan99**
parent issue : https://github.com/wso2/product-is/issues/16606
## Purpose
- Implement Unit Tests for  `org.wso2.carbon.identity.util`

- issue : https://github.com/wso2/product-is/issues/20139

### CODE REVIEW

File: component/org.wso2.carbon.identity.dpop/src/test/java/org/wso2/carbon/identity/dpop/test/helpers/DPoPProofUtil.java [hunk 1]
```diff
@@ -0,0 +1,157 @@
+package org.wso2.carbon.identity.dpop.test.helpers;
```
Comment (Thumimku) [inline component/org.wso2.carbon.identity.dpop/src/test/java/org/wso2/carbon/identity/dpop/test/helpers/DPoPProofUtil.java:1]: License? check other places

File: component/org.wso2.carbon.identity.dpop/src/test/java/org/wso2/carbon/identity/dpop/util/UtilsTest.java [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+        String dPoPProof = DPoPProofUtil.genarateDPoPProof();
+        String jwk = SignedJWT.parse(dPoPProof).getHeader().getJWK().toString();
+
+        spy(JWK.class);
+        when(JWK.parse(jwk)).thenReturn(mockJWK);
+        when(mockJWK.getKeyType()).thenReturn(new KeyType("some_type", Requirement.REQUIRED));
+        assertEquals(Utils.getThumbprintOfKeyFromDpopProof(dPoPProof), StringUtils.EMPTY);
+
+    }
+}
```
Comment (Thumimku) [inline component/org.wso2.carbon.identity.dpop/src/test/java/org/wso2/carbon/identity/dpop/util/UtilsTest.java:93]: New line

File: component/org.wso2.carbon.identity.dpop/pom.xml [hunk 3]
```diff
@@ -83,6 +83,38 @@
...
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>javax</groupId>
+            <artifactId>javaee-web-api</artifactId>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>com.h2database</groupId>
+            <artifactId>h2</artifactId>
```
Comment (Thumimku) [inline component/org.wso2.carbon.identity.dpop/pom.xml:115]: do we need it?

File: component/org.wso2.carbon.identity.dpop/src/test/java/org/wso2/carbon/identity/dpop/test/helpers/DPoPTestConstants.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
+package org.wso2.carbon.identity.dpop.test.helpers;
+
+public class DPoPTestConstants {
+
+    public static final String JWT_ACCESS_TOKEN =
```
Comment (Thumimku) [inline component/org.wso2.carbon.identity.dpop/src/test/java/org/wso2/carbon/identity/dpop/test/helpers/DPoPTestConstants.java:5]: do we need it?

File: component/org.wso2.carbon.identity.dpop/pom.xml (hunk 3 above)
Comment (PasinduLakshan99) [reply component/org.wso2.carbon.identity.dpop/pom.xml:115]: not at the moment..removed.

File: component/org.wso2.carbon.identity.dpop/src/test/java/org/wso2/carbon/identity/dpop/test/helpers/DPoPTestConstants.java (hunk 4 above)
Comment (PasinduLakshan99) [reply component/org.wso2.carbon.identity.dpop/src/test/java/org/wso2/carbon/identity/dpop/test/helpers/DPoPTestConstants.java:5]: not at the moment..removed.

File: component/org.wso2.carbon.identity.dpop/pom.xml [hunk 5]
```diff
@@ -83,6 +83,33 @@
...
+            <groupId>org.powermock</groupId>
+            <artifactId>powermock-module-testng</artifactId>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.powermock</groupId>
+            <artifactId>powermock-api-mockito</artifactId>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
```
Comment (Thumimku) [inline component/org.wso2.carbon.identity.dpop/pom.xml:108]: do we need it?

File: component/org.wso2.carbon.identity.dpop/src/test/java/org/wso2/carbon/identity/dpop/util/DPoPProofUtil.java [hunk 6]
```diff
@@ -0,0 +1,180 @@
+/*
+ * Copyright (c) 2023, WSO2 LLC. (https://www.wso2.com) All Rights Reserved.
```
Comment (Thumimku) [inline component/org.wso2.carbon.identity.dpop/src/test/java/org/wso2/carbon/identity/dpop/util/DPoPProofUtil.java:2]: License is not correct.

File: component/org.wso2.carbon.identity.dpop/src/test/java/org/wso2/carbon/identity/dpop/util/DPoPProofUtil.java [hunk 7]
```diff
@@ -0,0 +1,180 @@
...
+            "MEECAQAwEwYHKoZIzj0CAQYIKoZIzj0DAQcEJzAlAgEBBCAfBD6WwWEgj5FQ01/2zqR4NNu/i"+
+            "MEB/6fwhaMMh3GQFw==";
+
+    private static final String DPOP_JWT_TYPE = "dpop+jwt";
+
+    public static final String EC_DPOP_JWK_THUMBPRINT = "C07a9MZgz5wYywPc39Tw81gE8QzhkpC14sjx-2pAwbI";
+
+    public static final String RSA_DPOP_JWK_THUMBPRINT = "_Z3DHS03lCZVeRs-J9fO7JHuTE0BmVYuBF6Rdc5qjII";
+
+    //generate DPoP proof without passing any parameters
```
Comment (Thumimku) [inline component/org.wso2.carbon.identity.dpop/src/test/java/org/wso2/carbon/identity/dpop/util/DPoPProofUtil.java:99]: Improve the method

**Pull request closed without merging**